src/app/forms/validators/custom-validators.ts

```diff
@@ -133,4 +136,25 @@ export class CustomValidators {
   static invalidOption: ValidatorFn = (control: AbstractControl): ValidationErrors | null => {
     return '[INVALID_OPTION]' === control.value ? { invalidOption: true } : null;
   };
+
+  static resultDependantOnCustomDefects(store: Store<TestResultsState>): AsyncValidatorFn {
```

Comment:
Also, there is no unit test for the validator you've added. The async validators could also be put in a separate file/class if they have a different enum and map, might as well.

Reply:
All done, nice suggestion.

Reply:
All done, nice suggestion.